vllm/model_executor/layers/logits_processor.py

```diff
@@ -42,7 +42,9 @@ def __init__(self,
         # Soft cap the logits. Used in Gemma 2.
         self.soft_cap = soft_cap
         # Whether to use gather or all-gather to gather the logits.
-        self.use_gather = not current_platform.is_tpu()
+        # note: we import VLLM_USE_V1 dynamically to handle patching
+        from vllm.envs import VLLM_USE_V1
```

Comment:
`vllm.envs` has a module level `__getattr__` , and the recommended way is to use `envs.VLLM_USE_V1` . I don't know if `from vllm.envs import` works.